gateway/handlers/reader_test.go

```diff
@@ -1,4 +1,4 @@
-package tests
+package handlers
```

Comment:
I'm not sure that I like the style of putting tests inside the Go packages.

Reply:
Ok. It allows you to view how many statements are covered in the package. There may be another way to do this however. It's also convention in a lot of go projects. See https://golang.org/src/net/http/ and https://github.com/moby/moby/tree/master/client

Reply:
We should still propose changes that are counter to the existing conventions in the project. If that's something we decide to do we'd need to do it across the board.

Reply:
Yep